src/extensions/default/InlineColorEditor/main.js

```diff
@@ -39,41 +41,43 @@ define(function (require, exports, module) {
      * @return {?{color:String, marker:TextMarker}}
      */
     function prepareEditorForProvider(hostEditor, pos) {
-        var colorRegEx, cursorLine, match, sel, start, end, endPos, marker;
-
-        sel = hostEditor.getSelection();
-        if (sel.start.line !== sel.end.line) {
-            return null;
-        }
-
-        colorRegEx = new RegExp(ColorUtils.COLOR_REGEX);
+        var cursorLine, cssPropertyName, marker, endPos, end, semiColon, colorValue, hash, colon;
         cursorLine = hostEditor.document.getLine(pos.line);
 
-        // Loop through each match of colorRegEx and stop when the one that contains pos is found.
-        do {
-            match = colorRegEx.exec(cursorLine);
-            if (match) {
-                start = match.index;
-                end = start + match[0].length;
+        // Make a copy of cursorLine after removing spaces and ":" so that we can check for it in properties
+        cssPropertyName = cursorLine.split(':')[0].trim();
+
+        if (properties[cssPropertyName]) {
+            if (properties[cssPropertyName].type === "color") {
+                colon = cursorLine.indexOf(":");
+                semiColon = cursorLine.indexOf(";");
+                if (semiColon !== -1) {
+                  hash = cursorLine.indexOf("#");
+                  if (hash !== -1) {
+                    pos.ch = hash;
+                    colorValue = cursorLine.substring(hash, semiColon);
+                  } else {
+                    var colorString = cursorLine.substring(colon, semiColon);
+                    var firstCharacter = colorString.search(/\w/);
+                    pos.ch = colon + firstCharacter;
+                    colorValue = cursorLine.substring(colon + firstCharacter, semiColon);
+                  }
+                  endPos = {line: pos.line, ch: semiColon};
+                } else {
+                  pos.ch =  colon + 1;
+                  endPos = {line: pos.line, ch: cursorLine.length};
+                  colorValue = "white";
```

Comment:
If `white` is your default colour value, make it a top-level variable in the module, e.g., `var DEFAULT_COLOR = "white";` and use that here.